workflows/hicar.nf

```diff
@@ -231,227 +286,389 @@ workflow HICAR {
+        ch_multiqc_files = ch_multiqc_files.mix(HIPEAK.out.mqc.collect().ifEmpty([]))
+        ch_tfea_bed = ch_tfea_bed.mix(HIPEAK.out.fragmentPeak.map{[it[0], 'R1', it[1]]})
+        ch_tfea_bed = ch_tfea_bed.mix(HIPEAK.out.bed4tfea)
     }
 
     //
-    // Create igv index.html file
+    // Differential analysis
     //
-    ch_trackfiles.collect{it.join('\t')}
-        .flatten()
-        .collectFile(
-            name     :'track_files.txt',
-            storeDir : params.outdir+'/'+RelativePublishFolder.getPublishedFolder(workflow, 'IGV'),
-            newLine  : true, sort:{it[0]})
-        .set{ igv_track_files }
-    //igv_track_files.view()
-    IGV(igv_track_files, PREPARE_GENOME.out.ucscname, RelativePublishFolder.getPublishedFolder(workflow, 'IGV'))
+    if(!params.skip_diff_analysis){
+        if(params.da_tool=='hicexplorer'){
+            DA(
+                HICEXPLORER_CHICVIEWPOINT.out.interactions,
+                []
+            )
+        }else{
+            DA(
+                ch_de_files,
+                COOLER.out.samplebedpe
+            )
+        }
+        ch_versions = ch_versions.mix(DA.out.versions.ifEmpty(null))
+        ch_multiqc_files = ch_multiqc_files.mix(DA.out.mqc.collect().ifEmpty([]))
+        ch_annotation_files = ch_annotation_files.mix(DA.out.anno.map{[params.da_tool+'/'+it[1], it[2]]})
+        ch_v4c_files = ch_v4c_files.mix(DA.out.anno.map{[it[0], it[2]]})
+    }
+
+    //
+    // Motif analysis: for R2 reads and R1 reads
+    //
+    if(params.do_tfea){
+        if(params.tfea_tool=='atacseqtfea'){
+            ch_tfea_bed = ch_tfea_bed.map{[[id:it[1]], it[2]]}.groupTuple(by:0)
+                            .combine(
+                                SAMTOOLS_MERGE.out.bam
+                                        .map{[it[1]]}
+                                        .collect().toList())
+            ch_tfea_additional = PREPARE_GENOME.out.ucscname
+                                    .combine(PREPARE_GENOME.out.fasta)
+                                    .combine(PREPARE_GENOME.out.gtf)
+        }
+        TFEA(ch_tfea_bed, ch_tfea_additional)
+        ch_versions = ch_versions.mix(TFEA.out.versions.ifEmpty(null))
+    }
 
     //
-    // Annotate the MAPS peak
+    // Annotation
     //
     if(!params.skip_peak_annotation){
-        MAPS_PEAK.out.peak //[]
-            .map{meta, bin_size, peak -> [bin_size, peak]}
-            .filter{ it[1].readLines().size > 1 }
-            .groupTuple()
-            .set{ch_maps_anno}
-        BIOC_CHIPPEAKANNO_MAPS(ch_maps_anno, PREPARE_GENOME.out.gtf)
-        ch_versions = ch_versions.mix(BIOC_CHIPPEAKANNO_MAPS.out.versions.ifEmpty(null))
-        ch_multiqc_files = ch_multiqc_files.mix(BIOC_CHIPPEAKANNO_MAPS.out.png.collect().ifEmpty([]))
-        if(params.virtual_4c){
-            BIOC_CHIPPEAKANNO_MAPS.out.csv
+        BIOC_CHIPPEAKANNO(ch_annotation_files.combine(ATAC_PEAK.out.mergedpeak), PREPARE_GENOME.out.gtf, params.maps_3d_ext)
```

Comment:
As a suggestion, it is easier to understand if you add each input in a new line and use the appropriate indentation.